### PR TITLE
MMCA-4935-update cash account v2 to display closing balance in a new row

### PR DESCRIPTION
**Pull request by AdamCoops**

### CODE REVIEW

**Comment by tyagikumarvineet** [comment]
After these changes, values for some Withdrawals are not being displayed properly. Instead of showing 
-£200.50 its showing 

```
-
£200.50
```

**Comment by tyagikumarvineet** [comment]
Actual
Expected

File: app/views/components/daily_statements_v2.scala.html [hunk 1]
```diff
@@ -47,7 +47,7 @@
                                <span class="hmrc-responsive-table__heading" aria-hidden="true">
                                    ${messages("cf.cash-account.detail.transaction-type")}
                                </span>
-                                ${if(statement.transactionType.mrnLink.isDefined) statement.transactionType.mrnLink.get else statement.transactionType.textString.getOrElse(emptyString)}
+                               ${statement.transactionType.flatMap(_.mrnLink).getOrElse(statement.transactionType.flatMap(_.textString).getOrElse(emptyString))}
```
Comment (Neozxc) [inline app/views/components/daily_statements_v2.scala.html:50]: Could we please put this .getOrElse on another line? Something like below or how you like it

```suggestion
${statement.transactionType.flatMap(_.mrnLink)
.getOrElse(statement.transactionType.flatMap(_.textString).getOrElse(emptyString))}
```

File: app/viewmodels/CashAccountDailyStatementsViewModel.scala [hunk 2]
```diff
@@ -76,14 +76,19 @@ object CashAccountDailyStatementsViewModel {
           populateViewModelFromDeclarations(date, dStat.declarations)
 
         val declarationDailyStatementViewModelWithAccBalance = declarationDailyStatementViewModel.zipWithIndex.map {
-          case (dStatViewModel, 0) => dStatViewModel.copy(balance = Some(formatCurrencyAmount(dStat.closingBalance)))
           case (dStatViewModel, _) => dStatViewModel
         }
 
+        val closingBalanceOnlyRow = DailyStatementViewModel(
+          date = dateAsDayMonthAndYear(dStat.date),
+          balance = Some(formatCurrencyAmount(dStat.closingBalance))
```
Comment (Neozxc) [inline app/viewmodels/CashAccountDailyStatementsViewModel.scala:84]: Could we please move paranthesis on line 85 to line 84?

**Review by tyagikumarvineet — APPROVED** [review]

**Review by Neozxc — APPROVED** [review]

**Pull request merged**